zerver/lib/test_runner.py

```diff
@@ -302,9 +304,14 @@ def init_worker(counter: Synchronized) -> None:
     create_test_databases(_worker_id)
 
     # Every process should upload to a separate directory so that
-    # race conditions can be avoided.
-    settings.LOCAL_UPLOADS_DIR = '{}_{}'.format(settings.LOCAL_UPLOADS_DIR,
-                                                _worker_id)
+    # race conditions can be avoided.  By setting `TEST_RUN_DIR`
+    # here, each child process's sub-directory looks like the
+    # root for any given subsuite running in a particular worker.
+    worker_path = os.path.join(settings.TEST_RUN_DIR, 'worker_{}'.format(_worker_id))
+    os.makedirs(worker_path, exist_ok=True)
+    settings.TEST_RUN_DIR = worker_path
```

Comment:
The structure you have in this first commit of resetting `TEST_RUN_DIR` repeatedly is super confusing.  

I think we just should have two variable names:
* `TEST_RUN_DIR` as the root directory.
* `TEST_WORKER_DIR` as the per-worker directory.

Reply:
Right, updated.